sandbox/src/Http/RequestBuilder.php

```diff
@@ -0,0 +1,217 @@
+            $anchors
+        ));
+    }
+
+    public function setGivenNames($value, $optional = 'false', array $anchors = [])
+    {
+        $this->addAttribute($this->createAttribute(
+            Profile::ATTR_GIVEN_NAMES,
+            $value,
+            '',
+            $optional,
+            $anchors
+        ));
+    }
+
+    public function setDateOfBirth($value, $optional = 'false', array $anchors = [])
+    {
+        $this->addAttribute($this->createAttribute(
+            Profile::ATTR_DATE_OF_BIRTH,
+            $value,
+            '',
+            $optional,
+            $anchors
+        ));
+    }
+
+    public function setGender($value, $optional = 'false', array $anchors = [])
+    {
+        $this->addAttribute($this->createAttribute(
+            Profile::ATTR_GENDER,
+            $value,
+            '',
+            $optional,
+            $anchors
+        ));
+    }
+
+    public function setNationality($value, $optional = 'false', array $anchors = [])
+    {
+        $this->addAttribute($this->createAttribute(
+            Profile::ATTR_NATIONALITY,
+            $value,
+            '',
+            $optional,
+            $anchors
+        ));
+    }
+
+    public function setPhoneNumber($value, $optional = 'false', array $anchors = [])
+    {
+        $this->addAttribute($this->createAttribute(
+            Profile::ATTR_PHONE_NUMBER,
+            $value,
+            '',
+            $optional,
+            $anchors
+        ));
+    }
+
+    public function setSelfie($value, $optional = 'false', array $anchors = [])
+    {
+        $base64Selfie = base64_encode($value);
+        $this->setBase64Selfie($base64Selfie, $optional, $anchors);
+    }
+
+    public function setBase64Selfie($value, $optional = 'true', array $anchors = [])
+    {
+        $this->addAttribute($this->createAttribute(
+            Profile::ATTR_SELFIE,
+            $value,
+            '',
+            $optional,
+            $anchors
+        ));
+    }
+
+    public function setEmailAddress($value, $optional = 'false', array $anchors = [])
+    {
+        $this->addAttribute($this->createAttribute(
+            Profile::ATTR_EMAIL_ADDRESS,
+            $value,
+            '',
+            $optional,
+            $anchors
+        ));
+    }
+
+    public function setPostalAddress($value, $optional = 'false', array $anchors = [])
+    {
+        $this->addAttribute($this->createAttribute(
+            Profile::ATTR_POSTAL_ADDRESS,
+            $value,
+            '',
+            $optional,
+            $anchors
+        ));
+    }
+
+    public function setStructuredPostalAddress($value, $optional = 'false', array $anchors = [])
+    {
+        $this->addAttribute($this->createAttribute(
+            Profile::ATTR_STRUCTURED_POSTAL_ADDRESS,
+            $value,
+            '',
+            $optional,
+            $anchors
+        ));
+    }
+
+    public function setDocumentDetails($value, $optional = 'true', array $anchors = [])
```

Comment:
I'm not clear what type the _$value_ parameter should be.

We need to provide users with a fool-proof way to create this test value - I don't want to trust them to give us a correctly formatted string.  Ideally, we give them an easy, safe way to build a well-formed _DocumentDetails_ instance and then they can pass that to us.  We will convert that object to a string for them, because
1. we know what that format should be
2. if there is any problem with the structure, we can check it before we even bother sending it to _connect_.  Should be easier to identify problems that way.